From 8a93fe813ff69bcc176781c7cebe5a9c0c5fc252 Mon Sep 17 00:00:00 2001
From: Niels de Vos <ndevos@redhat.com>
Date: Wed, 10 Dec 2014 00:09:50 +0100
Subject: [PATCH 35/35] fill the cache when calling fuse_send_write_pages()

---
 fs/fuse/file.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/fuse/file.c b/fs/fuse/file.c
index d4baa4f..7da9ab6 100644
--- a/fs/fuse/file.c
+++ b/fs/fuse/file.c
@@ -1072,6 +1072,8 @@ static size_t fuse_send_write_pages(struct fuse_req *req, struct file *file,
 			count = 0;
 		offset = 0;
 
+		fuse_readpage_to_fscache(inode, page);
+
 		unlock_page(page);
 		page_cache_release(page);
 	}
-- 
1.9.3

